src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PersonNameTest.java

```diff
@@ -10,62 +10,37 @@
-            .firstForename("CHARLES")
-            .title("MR")
-            .secondForename("ANTHONY")
-            .build();
-
-        final String actual = personName.toEdifact();
-        assertEquals(expected, actual);
-    }
-
-    @Test
-    void testToEdifactWithTypeOnlyReturnsCorrectValue() {
-        final String expected = "PNA+PAT+T247:OPI'";
-
-        final PersonName personName = PersonName.builder()
-            .nhsNumber("T247")
-            .patientIdentificationType(PatientIdentificationType.OFFICIAL_PATIENT_IDENTIFICATION)
-            .build();
-
-        final String actual = personName.toEdifact();
-        assertEquals(expected, actual);
-    }
-
-    @Test
-    void testToEdifactWithBlankIdentificationAndBlankNamesThrowsException() {
-        final PersonName personName = PersonName.builder().build();
-
-        final EdifactValidationException exception = assertThrows(EdifactValidationException.class,
-            personName::toEdifact);
-
-        assertEquals("PNA: At least one of patient identification and person name details are required",
-            exception.getMessage());
-    }
-
     @Test
     void testFromString() {
-        assertAll("fromString",
-            () -> assertEquals(ID_ONLY_VALUE, PersonName.fromString(ID_ONLY).getValue()),
-            () -> assertEquals(ID_AND_NAMES_VALUE,
-                PersonName.fromString(ID_AND_NAMES).getValue()),
-            () -> assertEquals(NAMES_ONLY_VALUE, PersonName.fromString(NAMES_ONLY).getValue()));
+        var fromStringIdOnly = PersonName.fromString(ID_ONLY);
+        var fromStringIdAndNames = PersonName.fromString(ID_AND_NAMES_VALUE);
+        var fromStringNamesOnly = PersonName.fromString(NAMES_ONLY_VALUE);
+
+        assertAll("fromStringIdOnly",
+            () -> assertEquals(fromStringIdOnly.getNhsNumber(), "RAT56"),
+            () -> assertNull(fromStringIdOnly.getFirstForename()),
+            () -> assertNull(fromStringIdOnly.getSecondForename()),
+            () -> assertNull(fromStringIdOnly.getSurname()),
+            () -> assertNull(fromStringIdOnly.getTitle()));
+
+        assertAll("fromStringIdAndNames",
+            () -> assertEquals(fromStringIdAndNames.getNhsNumber(), "RAT56"),
+            () -> assertEquals(fromStringIdAndNames.getFirstForename(), "SARAH"),
+            () -> assertEquals(fromStringIdAndNames.getSecondForename(), "ANGELA"),
+            () -> assertEquals(fromStringIdAndNames.getSurname(), "KENNEDY"),
+            () -> assertEquals(fromStringIdAndNames.getTitle(), "MISS"));
+
+        assertAll("fromStringNamesOnly",
+            () -> assertNull(fromStringNamesOnly.getNhsNumber()),
+            () -> assertEquals(fromStringNamesOnly.getFirstForename(), "SARAH"),
+            () -> assertEquals(fromStringNamesOnly.getSecondForename(), "ANGELA"),
+            () -> assertEquals(fromStringNamesOnly.getSurname(), "KENNEDY"),
+            () -> assertEquals(fromStringNamesOnly.getTitle(), "MISS"));
```

Comment:
These look like separate tests. I recommend having them each in their own `@Test`